qiskit/visualization/pulse_v2/plotters/matplotlib.py

```diff
@@ -0,0 +1,201 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+# pylint: disable=invalid-name
+
+"""Matplotlib plotter API."""
+
+from typing import Optional
+
+import matplotlib
+import matplotlib.pyplot as plt
+import numpy as np
+from matplotlib.patches import Rectangle
+
+from qiskit.visualization.exceptions import VisualizationError
+from qiskit.visualization.pulse_v2 import core, drawings, types
+from qiskit.visualization.pulse_v2.plotters.base_plotter import BasePlotter
+
+
+class Mpl2DPlotter(BasePlotter):
+    """Matplotlib API for pulse drawer.
+
+    This plotter places canvas charts along y axis of 2D canvas with vertical offset.
+    Each chart is map to X-Y axis of the canvas.
+    """
+
+    def __init__(self,
+                 canvas: core.DrawerCanvas,
+                 axis: Optional[plt.Axes] = None):
+        """Create new plotter.
+
+        Args:
+            canvas: Configured drawer canvas object. Canvas object should be updated
+                with `.update` method before set to the plotter API.
+            axis: Matplotlib axis object. When `axis` is provided, the plotter updates
+                given axis instead of creating and returning new matplotlib figure.
+        """
+        super().__init__(canvas=canvas)
+
+        # calculate height of all charts
+        canvas_height = 0
+        for chart in self.canvas.charts:
+            if not chart.is_active and not self.canvas.formatter['control.show_empty_channel']:
+                continue
+            canvas_height += chart.vmax - chart.vmin
+
+        if axis is None:
+            fig_h = canvas_height * self.canvas.formatter['general.fig_chart_height']
+            fig_w = self.canvas.formatter['general.fig_width']
+
+            self.figure = plt.figure(figsize=(fig_w, fig_h))
+            self.ax = self.figure.add_subplot(1, 1, 1)
+        else:
+            self.figure = axis.figure
+            self.ax = axis
+
+        self.initialize_canvas()
+
+    def initialize_canvas(self):
+        """Format appearance of matplotlib canvas."""
+        self.ax.set_facecolor(self.canvas.formatter['color.background'])
+
+        # axis labels
+        self.ax.set_yticklabels([])
+        self.ax.yaxis.set_tick_params(left=False)
+
+    def draw(self):
```

Comment:
What will happen if this method is called twice?

Reply:
Plotter is initialized for each drawer call and thus draw is called only once. For dynamic image update we can use another plotter. If we call this twice with some hack perhaps all image components are drawn twice because I don't clear figure.